lib/spack/spack/store.py

```diff
@@ -70,11 +72,31 @@ def reindex(self):
 
 def _store():
     """Get the singleton store instance."""
-    root = spack.config.get('config:install_tree', default_root)
-    root = spack.util.path.canonicalize_path(root)
+    install_tree = spack.config.get('config:install_tree', {})
 
-    return Store(root,
-                 spack.config.get('config:install_path_scheme'),
+    if isinstance(install_tree, six.string_types):
+        tty.warn("Using deprecated format for configuring install_tree")
```

Comment:
Now that you have updated `spack config update` to automatically update this, I think it would be useful to mention that in the warning message (likewise for the warning about `install_path_scheme`)